includes/wcs-renewal-functions.php

```diff
@@ -35,7 +35,17 @@ function wcs_create_renewal_order( $subscription ) {
 
 	WCS_Related_Order_Store::instance()->add_relation( $renewal_order, $subscription, 'renewal' );
 
-	return apply_filters( 'wcs_renewal_order_created', $renewal_order, $subscription );
+	/**
+	 * Provides an opportunity to monitor, interact with and replace renewal orders when they
+	 * are first created.
+	 *
+	 * @param WC_Order        $renewal_order The renewal order.
+	 * @param WC_Subscription $subscription  The subscription the renewal is related to.
+	 */
+	$filtered_renewal_order = apply_filters( 'wcs_renewal_order_created', $renewal_order, $subscription );
+
+	// It is possible that a filter function will replace the renewal order with something else entirely.
+	return $filtered_renewal_order instanceof WC_Order ? $filtered_renewal_order : $renewal_order;
```

Comment:
Just wanted to chime in to say I ❤️ this change. I was looking at https://github.com/Automattic/woocommerce-subscriptions-core/issues/757 and I think that error wouldn't have occurred with this change.

I wonder if it would be worth to report these kind of failures somewhere. I'm thinking that if there are 2 or 3+ hooks which use this filter correctly, the changes they make would be overridden by 1 poorly written piece of code that returns a bool or null etc.

We don't have a good mechanism for doing this kind of thing but I'm thinking something like a WC log entry or something.